### DISCUSSION TITLE
What is the recommended way to poll data from a loader?

### DISCUSSION THREAD

**Discussion by blake-mealey** [remix-run/remix#998] [Q&A] [answered]
I need to poll my loader for some real-time data and I'm not sure what the recommended way to do this is. I'm currently doing this:

```ts
const submit = useSubmit();

useEffect(() => {
  const timer = setTimeout(() => {
    submit(null, { method: 'get', replace: true });
  }, 1000);
  return () => clearTimeout(timer);
});
```

Works great although I am getting this warning in the console every time it polls:

![image](https://user-images.githubusercontent.com/2432634/145697865-db26a1d8-179a-49e5-85f4-bf29d4910302.png)

**Comment by sergiodxa** [comment]
Check https://sergiodxa.com/articles/automatic-revalidation-in-remix where I explain how to do it.

**Reply by blake-mealey** [reply]
Basically the only difference between my implementation and yours is the different method we are calling (`submit` vs `navigate`). I'd like to verify that `navigate` is indeed the "better" one to use here and _why_ if possible.

**Reply by blake-mealey** [reply]
I just took a look at the code and it looks like `submit` eventually calls `navigate` so it makes sense to use `navigate` directly! I'll switch to `navigate`, thank you for the help :)

**Comment by grunklejp** [comment, marked as answer]
I believe the current best practice to do this now is with `useRevalidator` natively in remix:
https://remix.run/docs/en/main/hooks/use-revalidator

**Reply by blake-mealey** [reply]
Nice, thanks!

**Reply by manzaloros** [reply]
This solution doesn't scale well when your loader makes other expensive network calls that you only want to run once.

You would want to have a separate resource route that you'd poll on an interval with a fetcher so that your loader isn't run more than once.
